pep-0570.rst

```diff
@@ -385,6 +385,305 @@ as a rule is not LL(1). This is the reason the rule needs to be included in
+
+--------------------------------------
+When to use positional-only parameters
+--------------------------------------
+
+Sometimes, there are functions that take arguments that do not have semantic
+meaning, that is, the name of the argument does not carry information of what
+it is. For example, let's imagine we want to create a function to compute some
+math operation ::
+
+    def hilbert_multiply(x1, x2):
+        ....
+
+Here, the arguments ``x1`` and ``x2`` do not carry meaning on them: they serve
+as mere placeholders for two items that can be passed to the function. If the function
+is declared without positional only parameters, users can call the function using the
+argument names ::
+
+    >>> hilbert_multiply(x1=elem1, x2=elem2)
+
+But now, let's say that you want to clarify what ``x1`` and ``x2`` are, providing better
+names that will appear in the documentation and in the help section. So you would
+like to change the signature to ::
+
+
+    def hilbert_multiply(tensor1, tensor2):
+        ....
+
+Sadly, this is a backwards incompatible change, as users that were calling the function
+using the old names will receive this error message ::
+
+    >>> hilbert_multiply(x1=elem1, x2=elem2)
+    Traceback (most recent call last):
+      File "<stdin>", line 1, in <module>
+      TypeError: hilbert_multiply() got an unexpected keyword argument 'x1'
+
+To allow changing the names of the parameters in the future, you can mark them
+as positional-only. In this way users cannot pass the parameters by name so you
+can change them later ::
+
+    def hilbert_multiply(x1, x2, /):
+        ....
+
+Indeed, if users try to pass them using the names they will receive an error ::
+
+    >>> hilbert_multiply(x1=elem1, x2=elem2)
+    Traceback (most recent call last):
+      File "<stdin>", line 1, in <module>
+      TypeError: hilbert_multiply() got some positional-only arguments passed as keyword arguments: 'x1, x2'
+
+This is not the only situation where positional-only parameters can be useful.
+Let's say that you want your function to receive an indeterminate number of parameters,
+so the function will perform the operation with all of them, instead of calling the
+function multiple times with each pair. The signature will be now ::
+
+    def hilbert_multiply(*tensors):
+        ....
+
+Notice that if you used previously positional-only parameters, this is a backwards
+compatible change as the language is making sure that users were not using the name
+of the parameters when calling the function. So now, the old way of calling and the
+new way of calling are still possible ::
+
+    >>> hilbert_multiply(elem1, elem2) # Old way of calling the function
+
+    >>> hilbert_multiply(elem1, elem2, elem3, elem 4) # New way of calling the function
+
+If the function was not declared previously using positional-only parameters, then
+users will receive an error when calling the function as they used to do ::
```

Comment:
I'm going to change this to "as they used to", because "as they used to do" sounds wrong to me, but I think in BrE there are certain situations where it is idiomatic to say things like, "I would do" where Americans would say "I would" - I am not sure if this is one of them.